include/mega/megaclient.h

```diff
@@ -180,6 +183,31 @@ class MEGA_API FetchNodesStats
     dstime timeToTransfersResumed;
 };
 
+
+// Helper class for MegaClient.  
+// Maintains a small thread pool for executing independent operations such as encrypt/decrypt a block of data
+// The number of threads can be 0 (eg. for helper MegaApi that deals with public folder links) in which case something queued is 
+// immediately executed synchronously on the caller's thread
+struct MegaClientAsyncQueue
```

Comment:
I think this class will suit better in `utils.h`/`utils.cpp` rather than in MegaClient. It could be reused, in example, for the upcoming nodes on demand to load nodes from DB asynchronously.

Reply:
No problem moving it, will do.  I had planned for it to be generic like that initially, until realizing that it needed the SymmCipher as a member for this case, purely because those are very expensive to construct, but very cheap to reuse.   I'm sure there is a way to make it reusable, but I didn't want to further complicate things when it's only used for this case at this stage - anyway, will move it.  cheers